### DISCUSSION TITLE
support the command `__git_ps1` for git branch prompt?

### DISCUSSION THREAD

**Discussion by goyalyashpal** [msys2/MSYS2-packages#3512] [Q&A]
So, 
- the prompt `PS1` of gitforwindows contains this (command?) `__git_ps1` for (i think) fetching the name of the branch
- it will be nice to have that command available in msys `bash` too - even for still using the gitforwindows for git...
(as i don't use the "Git bash" launcher; rather terminals launch the system wide configured shell - msys bash)
- the cloud9's terminal too have this command in its `PS1`

### Examples of `PS1`

- gitforwindows prompt 
```console
$ echo $PS1
\[\033]0;$TITLEPREFIX:$PWD\007\]\n\[\033[32m\]\u@\h \[\033[35m\]$MSYSTEM \[\033[33m\]\w\[\033[36m\]`__git_ps1`\[\033[0m\]\n$

$ echo `__git_ps1`
(master)
```

<details>
<summary><tt>PS1</tt> Samples collected from shells elsewhere</summary>

(digress: the `$TITLEPREFIX:$PWD` seems to be similar to what's suggested in https://github.com/msys2/MSYS2-packages/issues/1798)

- MSYS launcher's prompt
```console
$ echo $PS1
\[\e]0;\w\a\]\n\[\e[32m\]\u@\h \[\e[35m\]$MSYSTEM\[\e[0m\] \[\e[33m\]\w\[\e[0m\]\n\$
```

- cloud9/ace terminal's prompt
```console
$ echo $PS1
\[\e]0;\a\033k$(cwdSlashAtEnd base)\033\\\]\[$(printf "\x0f")\033[01;34m\]$(cwdSlashAtEnd)\[\033[00m\]$(__git_ps1 " (%s)") $

$ echo $(cwdSlashAtEnd base)
~/

$ echo $(__git_ps1 " (%s)")
<did not have a git directory there>
```

- MS-gh-Codespace prompt
```console
$ echo $PS1
\[\]$(_prompt)\[\]
```

</details>

### Additional
* Duplicate searches: [git] | [prompt]

[git]: https://github.com/msys2/MSYS2-packages/discussions?discussions_q=git
[prompt]: https://github.com/msys2/MSYS2-packages/discussions?discussions_q=prompt

**Comment by Biswa96** [comment]
You could use [oh-my-zsh](https://ohmyz.sh/) configurations with zsh shell.

**Reply by goyalyashpal** [reply]
just to clarify: i am not saying for it to be necessarily be provided by default... i would be okay to be able to do it via some additional add-on package as well like the solution for pacboy was shared.

but zsh shell is likely entirely new system i guess, i have heard it's heavy.

**Reply by goyalyashpal** [reply]
found something on 🔍 : [__git_ps1](https://duckduckgo.com/?q=__git_ps1&t=vivaldi&ia=web)

* A simple command line prompt enhancer ...
https://github.com/watson/git-ps1

The shown syntax for this: `$(git-ps1 " (%s)")` looks similar to the cloud9's above: `$(__git_ps1 " (%s)")`

**Comment by KSR-Yasuda** [comment]
In my env, `__git_ps1` is a function defined in `/usr/share/git/completion/git-prompt.sh`, that is owned by `msys/git` package.
If you are just looking for the `__git_ps1` command, why don't you install `msys/git` package?

**Reply by goyalyashpal** [reply]
> why don't you install `msys/git` package?

excerpts from docs mentioned here:
- https://github.com/msys2/MSYS2-packages/issues/3358#issuecomment-1399326987
